xcore/base/xcam_buffer.h

```diff
@@ -103,6 +103,7 @@ struct _XCamVideoBuffer {
     XCamVideoBufferInfo   info;
     uint32_t              mem_type;
     int64_t               timestamp;
+    int                   hw_codec;
```

Comment:
can you use mem_type instead?
typedef enum {
    XCAM_MEM_TYPE_CPU,
    XCAM_MEM_TYPE_GPU,
    XCAM_MEM_TYPE_PRIVATE = 0x8000,
    XCAM_MEM_TYPE_PRIVATE_BO,
} XCamMemType;